### ISSUE TITLE
[BUG] Read Status not updated immediately on toggle.

### ISSUE THREAD

**Issue by BaccanoMob** [darylhjd/mangadesk#63]
Earlier Toggle Read gave an error (at the time the API was not fixed). At the moment the endpoint was supposedly fixed
![image](https://user-images.githubusercontent.com/82655889/147413998-ecaba69e-9e21-409d-a43b-3305175ed5d0.png)
We get no errors but Ctrl + R is unresponsive and does not toggle. I even checked via website that the read status of chapters I tested were not updated.

**Comment by darylhjd** [comment]
> We get no errors but Ctrl + R is unresponsive and does not toggle.

could you describe this more? (does the app freeze, does the table update, etc...)

my laptop died so i wont be able to diagnose this properly anytime soon.

**Comment by darylhjd** [comment]
if possible, you can also copy paste the content of the log file for the run where you experienced this problem (you can just paste all the content of the file)

**Comment by BaccanoMob** [comment]
> if possible, you can also copy paste the content of the log file for the run where you experienced this problem (you can just paste all the content of the file)

was about to ask the location but the correct location is %AppData%/Roaming/mangadesk/logs , logs was not in %AppData%/local 
and here you go

2021/12/26 17:50:14 logging.go:49: Session started at 2021-12-26 17-50-14
2021/12/26 17:50:15 main_page.go:40: Creating new main page...
2021/12/26 17:50:15 main_page.go:92: Using logged main page.
2021/12/26 17:50:15 service.go:27: Initialised starting screen.
2021/12/26 17:50:15 service.go:31: Running app...
2021/12/26 17:50:15 main_page.go:99: Setting logged grid...
2021/12/26 17:50:15 main_page.go:115: Setting logged table...
2021/12/26 17:50:15 main_page.go:110: Finished setting logged grid.
2021/12/26 17:50:15 main_page.go:204: Finished setting logged table.
2021/12/26 17:50:18 page_inputs.go:169: Selected row 7 on main page.
2021/12/26 17:50:18 manga_page.go:161: Setting up manga page chapter table...
2021/12/26 17:50:19 manga_page.go:295: Got 0 of 5 chapters
2021/12/26 17:50:22 modals.go:47: Removing toggle_read_chapters_modal modal
2021/12/26 17:50:24 modals.go:47: Removing toggle_read_chapters_modal modal

Apparently it only removes (i tried toggling twice like on off to see what happens)

**Comment by BaccanoMob** [comment]
It should mark "Y" for when read and remove "Y" when changed to unread

and I hope this helps solve the issue

**Comment by darylhjd** [comment]
thanks. seems like ive already forgotten Windows stuff.

> Apparently it only removes (i tried toggling twice like on off to see what happens)

so the app is able to toggle to unread but not to read? 

> It should mark "Y" for when read and remove "Y" when changed to unread

does the table update correctly?

**Comment by BaccanoMob** [comment]
I am really sorry, it apparently is working as intended while there is still a small bug.

So the mistake I made was i tried testing with single chapters. For example, I logged in and opened a manga page
![image](https://user-images.githubusercontent.com/82655889/147415797-ec66a378-5ec8-44bc-bfda-dca6388c9b56.png)

1. If press "Enter" as is, I get prompted if i should download the chapter(s). Meaning the highlighted chapter is selected.
2. If I instead press "Ctrl+R", I get a prompt but nothing happens. Meaning the highlighted chapter is not selected for some reason. This is the small bug.

While if i use "Ctrl+E" like 

1. for single selection (top chapter)
![image](https://user-images.githubusercontent.com/82655889/147415834-23d4d356-dfad-4140-8065-ee6e8dac0d75.png)

2. for multiple selection (top 3 chapters)
![image](https://user-images.githubusercontent.com/82655889/147415840-34f17c58-44fe-4c0c-92da-e2abda547e88.png)

it works but the changes are not reflected instantaneously. So if I toggle with the above case. After confirming the prompt, It still looks like this
![image](https://user-images.githubusercontent.com/82655889/147415885-69c384dd-4210-4ded-8fc9-6a314392c9c6.png)
but upon leaving and entering the page again it looks like (the changes are applied)
![image](https://user-images.githubusercontent.com/82655889/147415893-ca8474ef-be43-4f59-b77e-36a4516b7578.png)
Toggles correctly with the chapter on top as well

TLDR
It would be nice if it was updated instantaneously as seen with the downloads. 
And also for toggling only single chapters without Ctrl+E would be more convenient. 

Since you mentioned your laptop died, I tried putting as detailed as possible. Hope it helps.

**Comment by BaccanoMob** [comment]
also log states "Removing toggle_read_chapters_modal modal" whenever toggling whether marking as read or unread irrespective.

**Comment by darylhjd** [comment]
wow very detailed, thanks :)

i know what the problem is now haha, so ill work on it when i can!

the log is just showing the app removing the confirmation prompt (the modal), so no worries about that.

**Comment by darylhjd** [comment]
I'm trying to see if I can implement Ctrl+R toggle without selecting with Ctrl+E, however it seems unfeasible at the moment (due to library limitations).
